senders/test_server_test.go

```diff
@@ -71,3 +72,13 @@ func decodeMetricLines(request *http.Request) ([]string, error) {
 func (s *testServer) Close() {
 	s.httpServer.Close()
 }
+
+func (s *testServer) hasReceivedInternalMetric(metricLines []string, internalMetricName string) bool {
```

Comment:
rather than having a `metricsLines` parameter here, you can use `s.MetricLines` inside the method